### PR TITLE
Fix 2695: local open + let bindings

### PR DESCRIPTION
**Pull request by SanderSpies**
Solves #2695.

Makes it possible to use:

```
 let y = Promise.Ops.(
  let* x = Js.Promise.resolve(42);
  Js.Promise.resolve(x * 2);
);
```

Printing isn't perfect, but good enough for now.

### CODE REVIEW

**Review by anmonteiro — APPROVED** [review]
This looks great, thanks!

File: src/reason-parser/reason_pprint_ast.ml [hunk 1]
```diff
@@ -6442,7 +6442,12 @@ let printer = object(self:'self)
                     (atom (".")))
                   (self#formatNonSequencyExpression e))
           else
-            Some (makeLetSequence (self#letList x))
+            Some
```
Comment (anmonteiro) [inline src/reason-parser/reason_pprint_ast.ml:6445]: this looks exactly like the `then` branch.

I'm surprised this doesn't break `let open Foo; bar;` for example?

Perhaps the `letop` case should be added to `isSeriesOfOpensFollowedByNonSequencyExpression`, then we'd get printing for free?

Comment (SanderSpies) [reply src/reason-parser/reason_pprint_ast.ml]: Well, not exactly the same. The first branch results into extra braces: 
```
-      let* x = Js.Promise.resolve(42);
-      Js.Promise.resolve(x * 2);
+      {
+        let* x = Js.Promise.resolve(42);
+        Js.Promise.resolve(x * 2);
+      }
```

Comment (SanderSpies) [reply src/reason-parser/reason_pprint_ast.ml]: I did make a small change to avoid code duplication

Comment (anmonteiro) [reply src/reason-parser/reason_pprint_ast.ml]: I don't see how this is correct yet. I might be missing something.

Aren't we replacing ALL `open` calls with `Foo.( ...  )`?

Comment (SanderSpies) [reply src/reason-parser/reason_pprint_ast.ml]: @anmonteiro 

Not exactly, but the solution wasn't perfect (it didn't work for nested local open modules). I've tried to improve on it with help of an extra attribute set by the parser.

File: src/reason-parser/reason_pprint_ast.ml [hunk 2]
```diff
@@ -6433,15 +6458,17 @@ let printer = object(self:'self)
                 | _ -> Some (self#extension e)
           end
         | Pexp_open (me, e) ->
-            if self#isSeriesOfOpensFollowedByNonSequencyExpression x then
+            if self#isSeriesOfOpensFollowedByNonSequencyExpression x then (
```
Comment (SanderSpies) [inline src/reason-parser/reason_pprint_ast.ml:6461]: Todo: make the same as before.

**Comment by anmonteiro** [comment]
Thank you again, and sorry for the delay

**Pull request merged**